ios/RNCameraRoll.m

```diff
@@ -60,86 +85,96 @@ @implementation RNCameraRoll
         } else if (adjDate == nil && [epochNSDate timeIntervalSinceDate:creDate] > 0) {
           // if creation timestamp is less than our filter and the image has never been modified, return
           return;
-        } else if ( [extension caseInsensitiveCompare:@"heic"] == NSOrderedSame ) {
-          [self _processHEIC:asset orgFilename:orgFilename];
         } else {
-          [self _processJPG:asset orgFilename:orgFilename];
+          dispatch_group_enter(assetDatasGroup);
+          [self _writeToDisk:asset orgFilename:orgFilename onComplete:^(NSString *path, UIImageOrientation orientation) {
+            if (path) {
+              AssetData *data = [[AssetData alloc] init];
+              data.asset = asset;
+              data.path = path;
+              data.orientation = orientation;
+              [assetDatas addObject:data];
+            }
+            dispatch_group_leave(assetDatasGroup);
```

Comment:
Once the image is written to disk, record data about our asset and data on disk to the array where we're collecting these results. Leave the dispatch group.